modules/seqexec/server/src/main/scala/seqexec/server/SequenceConfiguration.scala

```diff
@@ -50,13 +51,33 @@ trait SequenceConfiguration {
       case kw         => StepState.Failed("Unexpected status keyword: " ++ kw)
     }.getOrElse(StepState.Failed("Logical error reading step status"))
 
-  def extractWavelength(config: CleanConfig): Option[Wavelength] =
+  def extractWavelength(config: CleanConfig): Either[SeqexecFailure, Wavelength] =
+    // Gmos uses Stringg
     config
       .extractAs[String](OBSERVING_WAVELENGTH_KEY)
-      .flatMap(v => Either.catchNonFatal(v.toDouble))
-      .map(Wavelength.fromMicrons[Double](_)).toOption
+      .flatMap(v => Either.catchNonFatal(v.toDouble).leftMap(_ => new ContentError(v)))
+      .map(Wavelength.fromMicrons[Double](_))
+      .orElse{
+        // GNIRS uses its own wavelength!!
+        config
+          .extractAs[GNIRSWavelength](OBSERVING_WAVELENGTH_KEY)
+          .map(w => Wavelength.fromMicrons(w.doubleValue()))
+      }.orElse{
+        // Maybe we use ocs Wavelength
+        config
+          .extractAs[Wavelength](OBSERVING_WAVELENGTH_KEY)
+      }.orElse{
+        // Just in case
+        config
+          .extractAs[java.lang.Double](OBSERVING_WAVELENGTH_KEY)
+          .map(v => Wavelength.fromMicrons[Double](v.doubleValue))
+      }.leftMap {
+        case _ =>
+          Unexpected(s"Error reading wavelength ${config.itemValue(OBSERVING_WAVELENGTH_KEY)}: ${config.itemValue(OBSERVING_WAVELENGTH_KEY).getClass()}")
+      }
```

Comment:
I was worried that we might store wavelength in different units for different instruments but I checked and that doesn't seem to be the case.  It's safe to assume that they are always microns.